src/by/epam/lab/DiscountProduct.java

```diff
@@ -0,0 +1,28 @@
+package by.epam.lab;
+
+public class DiscountProduct extends Product {
+    private final Byn discount;
+
+    public DiscountProduct() {
+        discount = new Byn();
+    }
+
+    public DiscountProduct(Byn discount) {
+        this.discount = discount;
+    }
+
+    public DiscountProduct(String name, Byn price, Byn discount) {
+        super(name, price);
+        this.discount = discount;
+    }
+
+    @Override
+    public Byn getPrice() {
+        return super.getPrice().sub(discount);
+    }
+
+    @Override
+    protected String fieldsToString() {
```

Comment:
I think last time mentor wanted us not to use string format because  it's slower